docs/reproducing_crashes.md

```diff
@@ -1,43 +1,183 @@
+$ make CC=clang LD=ld.lld -j$(nproc)
+```
+
+You also need a bootable disk image. Syzbot currently uses small Buildroot-based
+images that you can either [build locally](/tools/create-buildroot-image.sh) or
+[download](https://storage.googleapis.com/syzkaller/images/buildroot_amd64_2024.09.gz).
+
+Download and build the reproducer:
+```
+$ wget -O 'repro.c' 'https://syzkaller.appspot.com/x/repro.c?x=177beac0580000'
+$ gcc repro.c -lpthread -static -o repro
+```
+
+Run the VM:
+```
+$ export DISK_IMAGE='buildroot_amd64_2024.09'
+$ qemu-system-x86_64 -m 2G -smp 2,sockets=2,cores=1 -drive file=$DISK_IMAGE,format=raw -net nic,model=e1000 -net user,host=10.0.2.10,hostfwd=tcp::10022-:22 -enable-kvm -nographic -snapshot -machine pc-q35-7.1
+```
+
+Run the reproducer:
+```
+$ scp -P 10022 -o UserKnownHostsFile=/dev/null  -o StrictHostKeyChecking=no -o IdentitiesOnly=yes ./repro root@127.0.0.1:/root/
+$ ssh -p 10022 -o UserKnownHostsFile=/dev/null  -o StrictHostKeyChecking=no -o IdentitiesOnly=yes root@127.0.0.1 'chmod +x ./repro && ./repro'
+```
+
+
+## Using a Syz reproducer
+
+Syzkaller always generates a "Syz" reproducer first (in [Syzkaller
+DSL](/docs/program_syntax.md)). Afterwards, syzkaller attempts to convert the
+Syz reproducer into C code. The process does not always succeed due to the
+differences between the `syz-executor` environment and the environment emulated
+in the C reproducer. Therefore, in some cases, only the Syz version is
+available.
+
+To run a Syz reproducer locally, the required actions are mostly similar to
+those in the previous section.
+
+Download and [build](/docs/linux/setup.md#go-and-syzkaller) syzkaller. If you
+have Docker installed, the instructions are simpler:
+```
+$ git clone https://github.com/google/syzkaller.git
+$ ./tools/syz-env make
+```
+
+Build the kernel and boot the VM as described above.
+
+Download and run the reproducer:
+```
+$ export SYZKALLER_PATH="~/syzkaller"
+$ wget -O 'repro.syz' 'https://syzkaller.appspot.com/x/repro.syz?x=137beac0580000'
+$ scp -P 10022 -o UserKnownHostsFile=/dev/null  -o StrictHostKeyChecking=no -o IdentitiesOnly=yes $SYZKALLER_PATH/bin/linux_amd64/* ./repro.syz root@127.0.0.1:/root/
+$ ssh -p 10022 -o UserKnownHostsFile=/dev/null  -o StrictHostKeyChecking=no -o IdentitiesOnly=yes root@127.0.0.1 './syz-execprog -enable=all -repeat=0 -procs=6 ./repro.syz'
+```
+
+## Using ktest
+
+[ktest](https://evilpiepirate.org/git/ktest.git/tree/README.md) is a collection
+of tests for Linux and an infrastructure that simplifies running them locally.
+
+Ktest includes a special `syzbot-repro.ktest` test that automates building the
+kernel, booting the VM, fetching syzbot bug report details and running the
+reproducer.
+
+**Installation instructions:**
+```
+$ git clone git://evilpiepirate.org/ktest.git
+$ cd ktest
+$ export KTEST_PATH=$(pwd)
+$ sudo ./root_image init
+$ sudo ./root_image create
+$ sudo apt-get install capnproto
+$ cargo install --path /tmp/ktest
```

Comment:
Cap'n proto, and building & installing the rust binaries, shouldn't be needed for running ktest locally - those are just for the CI.

Reply:
The test requires `syzbot-get`
```
/tmp/ktest/tests/syzbot-repro.ktest: line 21: syzbot-get: command not found
```

To build `syzbot-get`, I run `cargo install --path /tmp/ktest` and get:

```
error: failed to run custom build command for `ci_cgi v0.1.0 (/tmp/ktest)`

Caused by:
  process didn't exit successfully: `/tmp/ktest/target/release/build/ci_cgi-c56bddee53fee4a6/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at src/build.rs:8:10:
  compiling schema: Error { kind: Failed, extra: "Failed to execute `capnp --version`: No such file or directory (os error 2). Please verify that version 0.5.2 or higher of the capnp executable is installed on your system. See https://capnproto.org/install.html" }
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `ci_cgi v0.1.0 (/tmp/ktest)`, intermediate artifacts can be found at `/tmp/ktest/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

Once `capnproto` is installed, the problem goes away. Though I think I can remove the `apt-get install` step from the documentation anyway -- `capnproto` is not the only one dependency and the errors always indicate what exactly is missing.